docs/user-interface/handlers/customize.md

```diff
@@ -1,122 +1,316 @@
 
-The following example customizes the background color of every control in the app, on Android, to cyan:
+In addition to these events, each control also has an overridable `OnHandlerChanged` method that's invoked when the `HandlerChanged` event fires, and a `OnHandlerChanging` method that's invoked when the `HandlerChanging` event fires.
+
+## Customize a control with a mapper
+
+The .NET MAUI `Entry` is a single-line text input control, that implements the `IEntry` interface. On iOS, the `EntryHandler` maps the `Entry` to an iOS `UITextField`. On Android, the `Entry` is mapped to an `AppCompatEditText`, and on Windows the `Entry` is mapped to a `TextBox`:
+
+:::image type="content" source="customize-images/entry-handler.png" alt-text="Entry handler architecture." border="false":::
+
+The `EntryMapper`, in the `EntryHandler` class, maps the cross-platform control API to the native control API. This mapper can be modified to customize the control on each platform:
 
 ```csharp
 using Microsoft.Maui;
 using Microsoft.Maui.Controls;
+using Microsoft.Maui.Graphics;
 
-public partial class App : Application
+namespace HandlersDemos.Views
 {
-    public App()
+    public partial class CustomizeEntryPage : ContentPage
     {
-        InitializeComponent();
+        public CustomizeEntryPage()
+        {
+            InitializeComponent();
 
+            Microsoft.Maui.Handlers.EntryHandler.EntryMapper["MyCustomization"] = (handler, view) =>
+            {
 #if __ANDROID__
-        Microsoft.Maui.Handlers.ViewHandler.ViewMapper[nameof(IView.Background)] = (h, v) =>
-        {
-            (h.NativeView as Android.Views.View).SetBackgroundColor(Microsoft.Maui.Graphics.Colors.Cyan.ToNative());
-        };
+                handler.NativeView.SetBackgroundColor(Colors.Transparent.ToNative());
+#elif __iOS__
+                handler.NativeView.BorderStyle = UIKit.UITextBorderStyle.None;
+#elif WINDOWS
+                handler.NativeView.FontWeight = Microsoft.UI.Text.FontWeights.Thin;
 #endif
+            };
+        }
     }
 }
 ```
 
-### Remove Android Entry underline
+In this example, the `Entry` customization occurs in a page class. Therefore, all `Entry` controls on Android, iOS, and Windows will be customized once the `CustomizeEntryPage` is displayed. The following customization is performed by using compiler preprocessing directives:
+
+- On Android, the underline is removed from the `Entry`.
+- On iOS, the border is removed from the `Entry`.
+- On Windows, the thickness of the font in the `Entry` is set to thin.
+
+> [!IMPORTANT]
+> The index into a mapper is an arbitrary string and does not have to correspond to a property exposed by a type, or a property on a native control. For example, `["MyCustomization"]` can be specified as a mapper index, with any native control modification being performed as the customization.
```

Comment:
Question on this. The "MyCustomization" doesn't make sense in this note and in the previous code example. How does that map to an `Entry` control?

**UPDATE**
Looking at the code below in the next section, there is a type check: `if (view is MyEntry)` so this makes me think that ALL handlers are applied to every control when they're created. There is no system that automatically runs handler X when control X is created. I didn't get that impression until I got this far and the IMPORTANT note confused me and then I saw the following code. That may prove confusing.

Reply:
I *think* this is resolved from an earlier comment that you went back and edited.